internal/component/otelcol/internal/scheduler/scheduler.go

```diff
@@ -100,6 +129,7 @@ func (cs *Scheduler) Run(ctx context.Context) error {
 
 			level.Debug(cs.log).Log("msg", "scheduling components", "count", len(components))
 			components = cs.startComponents(ctx, host, components...)
+			cs.onResume()
```

Comment:
if the component was created with New() instead of NewPaused() then this will panic because on the first run you will try to unlock the mut that's not locked

Reply:
Now it will no longer panic as calls to Resume can be repeated.

When writing code we still need to take care to use the correct paused or resumed consumer, but the consequences of a mistake shouldn't be catastrophic.